### ISSUE TITLE
Issues publishing by direct db connection after java issue

### ISSUE THREAD

**Issue by bbrunet** [gbif/ipt#1221]
Sometime last month our system admin deleted the java version on our webserver that our IPT (v2.2) was configured with. This caused four of our resources to fail during automated monthly publishing beginning Oct. 1 with an error that reads along the lines of "...Error writing data file for mapping Darwin Core Archive...". Two of these failed publications occurred prior to upgrading the IPT to version 2.3.1 at the beginning of the month and two occurred afterwards (I hadn't noticed the issue until after upgrading), a version which incidentally also had inherent bugs associated with publishing via direct connection, so I subsequently upgraded to version 2.3.2. This caused the error message for the affected resources to change to a "...Java heap space..." issue which lead us to the assumed root of the problem (that the Java version was deleted).

Now that we have re-established the path to the Java version that we're using for the IPT, I've been able to publish the two remaining resources without issue, but a problem still persists for the four affected resources. I looked at the resources in the iptdata directory (from a backup of the iptdata dir prior to upgrading to v2.3.1) and it appears that the failed attempts at publishing resulted in the current dwc-a version being deleted to make room for the new version (we don't have archival mode turned on), so even though the IPT returned back to the current version after the two failed publications prior to upgrading, those current dwc archive are no longer available. Also, contrary to the write error message noted above, the archive for the pending version was still saved. Despite this, subsequent attempts to publish also continue to fail. 

There is slightly different handling of the dwc archives for the two resources that failed after upgrading to v.2.3.1, namely that the iptdata/resource directories for the two resources now have dwc-a archives for both the current and pending versions, but when attempting to publish manually nothing happens following the "Are you sure?" window.

I've also noticed that the current version and pending versions on the manage resources pages for the affected resources are not present in the version history on each of these resources' home pages. The two resources that failed prior to upgrading to v.2.3.1 indicate on their home pages that the most recent versions are actually two versions prior to the current versions indicated on their manage resource pages (eg. 1.3 versus 1.5, respectively). Similarily, the two resources that failed after upgrading to v2.3.1 also have discrepancies between the version history and the current version, except that the difference is only a single version (eg. 1.4 versus 1.5). The unaffected resources show all published versions as expected. 

In summary, now that the Java issue has been resolved, and confirmation that fixing this has restored successful publishing of the two unaffected resources, I am still unable to publish the four affected resources.

Any help resolving this/these issues would be greatly appreciated! 

Thanks,
Bryan Brunet
University of Alberta Museums

**Comment by kbraak** [comment]
Thanks for reporting your issue. 

Version 2.2 had a [bug](https://github.com/gbif/ipt/issues/1155) when archival mode was turned off. Version 2.2.1 would have fixed the problem you described above. 

Anyways, let's try to focus on why "subsequent attempts to publish (those 4 datasets) also continue to fail." 

Can you please a) make sure debugging mode is on (see Configure IPT settings page), then for only 1 of the 4 datasets, b) preview its mapping, and if that's working c) try to re-publish it. If that fails d) look at the Admin/Logs page to see if any obvious errors appear. If nothing obvious appears to you, please download the complete log file and email it to me (kbraak@gbif.org) so that I can take a look myself. 

Thanks

**Comment by bbrunet** [comment]
Hi Kyle,

Thanks for you quick response. I've followed your guidelines and inspected the log page, but only see two warnings and no errors. The warnings are:

"...[org.apache.struts2.components.ServletUrlRenderer] - No configuration found for the specified action: 'publishAll.do' in namespace: ''. Form action defaulting to 'action' attribute's literal value."

and

"...[org.apache.struts2.components.ServletUrlRenderer] - No configuration found for the specified action: 'create.do' in namespace: ''. Form action defaulting to 'action' attribute's literal value."

The same two warning occur for each of the affected resources.

I'll email you the log that was generated for just one of the resources in a few minutes.

Thanks,
Bryan

**Comment by kbraak** [comment]
Marking this issue as a duplicate of #1155. The other bug existing in this issue was addressed separately in issue #1223, which has now been fixed and will be included in the next minor release 2.3.3.

Users affected by this issue not able to wait for 2.3.3 to be released can work around it by following (these instructions)[https://github.com/gbif/ipt/issues/1223#issuecomment-152476525] provided in issue #1223.
